### ISSUE TITLE
Treat lambda as function in spaces_left_parentheses_linter

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#685]
In r-devel, `spaces_left_parentheses_linter` complains about the newly introduced lambda syntax where `\(x)` is equivalent with `function(x)`.

```r
> lintr::lint("\\(x) x + 1\n")                                                                                                                                                                                                                            
<text>:1:2: style: Place a space before left parenthesis, except in a function call.
\(x) x + 1
 ^
```

**Comment by AshesITR** [comment]
How does the parse data look like in r-devel?
`source_file$parsed_content$token` would be interesting.

Probably there is a new token that needs adding to https://github.com/jimhester/lintr/blob/20d37fa943e9627beba8fc24f2e216eca9e8c956/R/spaces_left_parentheses_linter.R#L17

**Comment by renkun-ken** [comment]
Please take a look at https://github.com/r-lib/xmlparsedata/issues/17 and https://github.com/r-lib/xmlparsedata/pull/18.

**Comment by AshesITR** [comment]
That means the fix on our side should add `"'\\\\'"` to the exceptions.

Do you want to submit a PR?

Such a test should currently fail on `r-devel`:

```r
if (getRversion() > "4.0.3") {
  expect_lint("\\(x) x + 1", NULL, spaces_left_parentheses_linter)
}
```

**Comment by renkun-ken** [comment]
I'll send a PR for this.

As for the test, would you like the suggestion at https://github.com/r-lib/xmlparsedata/pull/18#issuecomment-739296842?

**Comment by AshesITR** [comment]
I wouldn't hard-code the svn revision (current and future r-devel will always have at least this rev. no).
Checking `>= 4.1.0` instead of `> 4.0.3` is fine with me. Is the r-devel version set to `4.1.0`?

**Comment by renkun-ken** [comment]
Sure. 

>Is the r-devel version set to 4.1.0

Yes, it is already 4.1.0.

**Comment by kpagacz** [comment]
This can be closed according to this comment: https://github.com/jimhester/lintr/pull/686#issuecomment-856425543
